### ISSUE TITLE
Implement the CLI `assess_trained_models`

### ISSUE THREAD

**Issue by weitse-hsu** [wehs7661/AEV-PLIG-refined#6]
Here is a summary for @savvag44 of what should be done for the CLI `assess_trained_models` to make it more convenient to assess trained models. 

## Plan of attack
Overall, the CLI should be able to do the following:
- **Task 1**: Given a directory that contains one or more trained models and pickled outputs from `AEV-PLIG`, the CLI should be able to test each of the trained models on any test set (assuming available PyTorch files), as well as calculate the performance for the ensemble. 
- **Task 2**: The CLI should create a CSV file that saves the predictions on the test set from each of the trained models, and from the ensemble model. (We can discuss later to see if it would be helpful to save intermediate features as well.) 
- **Task 3**: The script should estimate the uncertainty of the performance for each trained model and for the ensemble model. It should provide the following two options to estimate the uncertainty by bootstrapping (e.g., via a flag `-bm`/`--bootstrap_method`):
  - If `args.bootstrap_method == 'bootstrap_metric'`, the CLI will use the CLI method described in the AEV-PLIG paper. For example, for a test set with $N$ protein families, we calculate the uncertainty of a weighted PCC (or any other correlation coefficient) for a trained model with the following steps:
    - **Step 1**: Run the trained model on the test set, from which one can get the correlation coefficients for different $k$ protein families, i.e., $c_1$, $c_2$, ..., $c_k$. These coefficients come with weights proportional to the number of entries in the corresponding protein families, denoted as $w_1$, $w_2$, ..., $w_k$. The weighted correlation coefficient is then calculated as $c=\sum_{i=1}^k c_i w_i / \sum_{i=1}^k w_k$
    - **Step 2**: We treat the set of correlation coefficients as a dataset of $k$ data points, drawing $k$ samples with replacements. These $k$ drawn samples have their own corresponding weights. We then used these correlation coefficients and weights, which compose of a synthetic dataset, to calculate a weighted correlation coefficient. We call this one bootstrap iteration. 
    - **Step 3**: We repeat Step 2 for a specified number of times $N$ to get $N$ weighted correlation coefficients, then report the standard deviation of these $N$ values as the uncertainty for $c$. 
  - **Note**: This method has currently been implemented in `calc_metrics.py` in the class `MetricCalculator` and used in `train_aev_plig.py`. In the AEV-PLIG paper, the authors only consider protein families with more than 10 members. This can be done by the `n_min` argument passed to `MetricCalculator`. In general, this method should be fine, but the fact that each family in the FEP benchmark do not have a large number of members and there are not that many protein families in the benchmark, boostrapping the correlation coefficients themselves may be unstable. 
  - If `args.bootstrap_method == 'boostrap_data'`, the CLI will boostrap the test set entries, which should be more rigorous than the other method. For example, for a test set of $N$ protein families with the number of family members being $n_1$, $n_2$, ..., $n_N$, we do the following:
    -  **Step 1**: In a bootstrap iteration, we generate a synthetic dataset by (1) drawing $n_i$ samples with replacement for the $i$-th protein family and (2) combining the $N$ datasets. We then apply the trained model on this synthetic dataset to get a weighted correlatoin coefficient. 
    - **Step 2**: To get the uncertainty of this weighted correlation coefficient, we repeat **Step 1** a large number of times (specified by the user) and report the standard deviation. 
  - **Note**: The predictions made by a trained model are deterministic, so we only need to run the trained model on the entire test set, construct a dictionary (or other dataset structure) to look up. Then, during bootstrapping, we refer to this dictionary for the corresponding prediction to calculate the performance. This method is supposedly more rigorous than the first method. 

## Additional notes
- To get familiar with how to write a CLI, you can check existing CLIs in the package, e.g., `train_aev_plig.py`. In general, in a CLI you will need a `main` function, a function to read in arguments (I usually name it as `initialize` in this project), and any functions that you would like use in your main function. You will also need to modify `pyproject.toml`: search `[project.scripts]` in the file, and add `assess_trained_models = "aev_plig.cli.assess_trained_models:main". For the CLI to take effect, you should reinstall the project by running `pip install -e .` in the root directory of the repo. It would be nice if you can follow the coding style of existing CLIs, e.g., using `utils.Logger` for logging). 
- To check if Task 1 is done properly, you can compare the values calculated by your CLIs with the ones in the log files of some o the previously done experiments, e.g., the values in `AEV_PLIG_training/HiQBind_full/redo/log_files/train_aev_plig.log`, where the full dataset of HiQBind was used for training and the FEP benchmark was used for testing.
